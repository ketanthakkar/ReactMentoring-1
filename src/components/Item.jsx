import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {Link} from 'react-router-dom';

const ItemWrapper = styled.div`
    width: 250px;
    margin: 20px 75px 20px 0;
    
    :nth-child(3n) {
        margin: 20px 0 20px 0;
    }
`;

const ItemImage = styled.img`
    width: 250px;
    cursor: pointer;
`;

const ItemInfo = styled.div`
    display: flex;
    align-items: flex-start;
    justify-content: space-between;
    margin-top: 5px;
`;

const ItemTitle = styled.span`
    flex-basis: 70%;
`;

const ItemDate = styled.span`
    border: 1px solid gray;
    border-radius: 3px;
    padding: 1px 3px;
`;

const ItemGenres = styled.div`
    color: gray;
    margin-top: 5px;
    font-size: 14px;
`;

class Item extends React.Component {
    render() {
        const {
            item,
        } = this.props;

        const {
            poster_path,
            title,
            release_date,
            genres,
        } = item;

        return (
            <ItemWrapper>
                <Link to={`/film/${item.id}`}>
                    <ItemImage
                        src={poster_path}
                    />
                </Link>
                <ItemInfo>
                    <ItemTitle>{title}</ItemTitle>
                    <ItemDate>{release_date && release_date.slice(0, release_date.indexOf('-'))}</ItemDate>
                </ItemInfo>
                <ItemGenres>{genres && genres.join(' & ')}</ItemGenres>
            </ItemWrapper>
        )
    }
}

Item.propTypes = {
    item: PropTypes.object,
};

Item.defaultProps = {
    item: {},
};

export default Item;
