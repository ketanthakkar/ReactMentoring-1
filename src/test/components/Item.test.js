import React from 'react';
import { shallow } from 'enzyme';

import Item from 'components/Item';

describe('Item', () => {
    const item = shallow(<Item />);

    it('should be like snapshot', () => {
        expect(item).toMatchSnapshot();
    });
});
