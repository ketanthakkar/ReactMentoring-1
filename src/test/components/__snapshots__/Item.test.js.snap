// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should be like snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    genres={Array []}
    item={Object {}}
    onItemClick={[MockFunction]}
    poster_path=""
    release_date=""
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Item__ItemImage
          onClick={[Function]}
          src={undefined}
        />,
        <Item__ItemInfo>
          <Item__ItemTitle />
          <Item__ItemDate />
        </Item__ItemInfo>,
        <Item__ItemGenres />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClick": [Function],
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Item__ItemTitle />,
            <Item__ItemDate />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Item__ItemImage
            onClick={[Function]}
            src={undefined}
          />,
          <Item__ItemInfo>
            <Item__ItemTitle />
            <Item__ItemDate />
          </Item__ItemInfo>,
          <Item__ItemGenres />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Item__ItemTitle />,
              <Item__ItemDate />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
